core/com.b2international.snowowl.core.rest/src/com/b2international/snowowl/core/rest/SnowOwlApiConfig.java

```diff
@@ -120,6 +127,53 @@ public ObjectMapper objectMapper() {
 		return objectMapper;
 	}
 	
+	@Bean
+	@Scope(scopeName = WebApplicationContext.SCOPE_REQUEST, proxyMode = ScopedProxyMode.TARGET_CLASS)
+	public ObjectMapper objectMapper(@Autowired HttpServletRequest request) {
+		if (objectMappers == null) {
+			objectMappers = new MapMaker().makeMap();
+			objectMappers.put("includeNonNullObjectMapper", createObjectMapper());
+			objectMappers.put("includeNonNullObjectMapperPretty", createObjectMapper()
+					.enable(SerializationFeature.INDENT_OUTPUT));
+			objectMappers.put("includeAllObjectMapper", createObjectMapper()
+					.setSerializationInclusion(Include.ALWAYS));
+			objectMappers.put("includeAllObjectMapperPretty", createObjectMapper()
+					.setSerializationInclusion(Include.ALWAYS)
+					.enable(SerializationFeature.INDENT_OUTPUT));
+		}
+
+		boolean includeNulls = extractBooleanQueryParameterValue(request, INCLUDE_NULL);
+		boolean pretty = extractBooleanQueryParameterValue(request, PRETTY);
+		
+		if (includeNulls && pretty) {
+			return objectMappers.get("includeAllObjectMapperPretty");
+		} else if (includeNulls && !pretty) {
+			return objectMappers.get("includeAllObjectMapper");
+		} else if (!includeNulls && pretty) {
+			return objectMappers.get("includeNonNullObjectMapperPretty");
+		} else if (!includeNulls && !pretty) {
+			return objectMappers.get("includeNonNullObjectMapper");
+		} else {
+			throw new NotImplementedException("Unexpected case that should not happen");
+		}
+	}
+
+	private boolean extractBooleanQueryParameterValue(HttpServletRequest request, String queryParameterKey) {
+		String[] values = request.getParameterMap().containsKey(queryParameterKey) ? request.getParameterMap().getOrDefault(queryParameterKey, null) : null;
```

Comment:
`getOrDefault` already checks for the existence of the key:

```suggestion
		String[] values = request.getParameterMap().getOrDefault(queryParameterKey, null);
```